nas_ports/freenas/py-middlewared/files/middlewared.in

```diff
@@ -26,15 +26,29 @@ middlewared_start() {
 	/sbin/ifconfig lo0 127.0.0.1 up
 	if checkyesno middlewared_debug; then
 		/usr/local/bin/tmux new-session -s middlewared -d
-		/usr/local/bin/tmux send -t middlewared "env LC_ALL=en_US.UTF-8 ${command} -f ${plugins_dirs_arg}" ENTER
+		/usr/local/bin/tmux send -t middlewared "env PATH=$PATH:/usr/local/sbin:/usr/local/bin LC_ALL=en_US.UTF-8 /usr/local/bin/middlewared -f ${plugins_dirs_arg}" ENTER
 	else
 		env PATH=$PATH:/usr/local/sbin:/usr/local/bin LC_ALL=en_US.UTF-8 ${command} -f -P ${pidfile} -r /usr/local/bin/middlewared -f ${plugins_dirs_arg} --log-handler=file
 	fi
 	LD_LIBRARY_PATH=/usr/local/lib /usr/local/bin/midclt -t 120 waitready
 }
 
+middlewared_stop() {
+	if checkyesno middlewared_debug; then
+		/usr/local/bin/tmux kill-session -t 'middlewared'
+	else
+		if [ -f "${pidfile}" ] ; then
+			echo 'Stopping middlewared.'
+			/bin/pkill -F "${pidfile}"
```

Comment:
This put us in a different behavior than the original stop which sends the kill signal and wait for the pid.

Perhaps we can use builtin rc.subr functions for this?